flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/plan/nodes/dataset/DataSetAggregate.scala

```diff
@@ -157,4 +161,41 @@ class DataSetAggregate(
       case _ => result
     }
   }
+
+  /**
+    * Dummy [[Row]] into a [[DataSet]] for result after map operations.
+    * @param mapOperator after which insert dummy records
+    * @param tableEnv [[BatchTableEnvironment]] for getting rel builder and type factory
+    * @tparam IN mapOperator input type
+    * @tparam OUT mapOperator output type
+    * @return DataSet of type Row is contains null literals for columns
+    */
+  private def dummyRow[IN,OUT](
```

Comment:
I would not call this "Dummy row" but "Null row". Dummy row sounds like a useless mock up, but our null row has a good purpose.